/** showDoc.c **/

/*
*|***************************************************************************|*
*|                                                                           |*
*|   Program: phred                                                          |*
*|   Version: 0.000925.c                                                     |*
*|                                                                           |*
*|   Copyright (C) 1993-2000 by Phil Green and Brent Ewing.                  |*
*|   All rights reserved.                                                    |*
*|                                                                           |*
*|   This software is a beta-test version of the phred package.              |*
*|   It should not be redistributed or used for any commercial               |*
*|   purpose, including commercially funded sequencing, without              |*
*|   written permission from the author and the University of                |*
*|   Washington.                                                             |*
*|                                                                           |*
*|   This software is provided ``AS IS'' and any express or                  |*
*|   implied warranties, including, but not limited to, the                  |*
*|   implied warranties of merchantability and fitness for a                 |*
*|   particular purpose, are disclaimed.  In no event shall                  |*
*|   the authors or the University of Washington be liable for               |*
*|   any direct, indirect, incidental, special, exemplary, or                |*
*|   consequential damages (including, but not limited to,                   |*
*|   procurement of substitute goods or services; loss of use,               |*
*|   data, or profits; or business interruption) however caused              |*
*|   and on any theory of liability, whether in contract, strict             |*
*|   liability, or tort (including negligence or otherwise)                  |*
*|   arising in any way out of the use of this software, even                |*
*|   if advised of the possibility of such damage.                           |*
*|                                                                           |*
*|   Portions of the code benefit from ideas due to Dave Ficenec,            |*
*|   LaDeana Hillier, Mike Wendl, and Tim Gleeson.  These are                |*
*|   indicated in the relevant source files.                                 |*
*|                                                                           |*
*|***************************************************************************|*
*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#ifdef ANSI_C
int showDoc( void )
#else
int showDoc()
#endif
{

  printf( "\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "PHRED Documentation\n" );
  printf( "-------------------\n" );
  printf( "\n" );
  printf( "1. Introduction.\n" );
  printf( "\n" );
  printf( "   Phred reads DNA sequencer trace data, calls bases, assigns quality\n" );
  printf( "   values to the bases, and writes the base calls and quality values to\n" );
  printf( "   output files.  Phred can read trace data from SCF, ABI model 373 and\n" );
  printf( "   377 DNA sequencer chromatogram, and MegaBACE ESD chromatograms files,\n" );
  printf( "   automatically detecting the file format, and whether the chromat\n" );
  printf( "   file was compressed by gzip or UNIX compress.  After calling bases,\n" );
  printf( "   phred writes the sequences to files in either FASTA format, the\n" );
  printf( "   format suitable for XBAP, PHD format, or the SCF format.  Quality\n" );
  printf( "   values for the bases are written to FASTA format files or PHD files,\n" );
  printf( "   which can be used by the phrap sequence assembly program in order to\n" );
  printf( "   increase the accuracy of the assembled sequence.\n" );
  printf( "\n" );
  printf( "   Significant differences in this release\n" );
  printf( "\n" );
  printf( "   New\n" );
  printf( "\n" );
  printf( "     - add X86_GCC_LINUX definition. Defining this (in the supplied\n" );
  printf( "       Makefile) when compiling phred on x86 Linux machines using\n" );
  printf( "       the GCC compiler, causes phred to call bases and assign quality\n" );
  printf( "       values that are identical to those called and assigned when\n" );
  printf( "       phred runs on IEEE conforming UNIX machines. See the INSTALL\n" );
  printf( "       and Makefiles in this distribution and the Linux system file\n" );
  printf( "       /usr/include/fpu_control.h for additional information.\n" );
  printf( "\n" );
  printf( "     - add LI-COR quality value lookup tables for chromatograms created\n" );
  printf( "       by the LI-COR sequencing machine. See the section below entitled\n" );
  printf( "       LI-COR Data.\n" );
  printf( "\n" );
  printf( "   Modified\n" );
  printf( "\n" );
  printf( "     - set `source' string, which is written in the output SCF files,\n" );
  printf( "       based on machine type inferred from primer ID and phredpar.dat\n" );
  printf( "       rather than based on the chromatogram type. The possible `source'\n" );
  printf( "       values are\n" );
  printf( "\n" );
  printf( "         \"ABI 373A or 377\"\n" );
  printf( "         \"MegaBACE\"\n" );
  printf( "         \"ABI 3700\"\n" );
  printf( "         \"LI-COR 4000\"\n" );
  printf( "         \"unknown\"\n" );
  printf( "\n" );
  printf( "     - add `-trim_fasta', `-trim_scf', and `-trim_phd' options, which\n" );
  printf( "       cause phred to write trimmed sequence and quality values to\n" );
  printf( "       FASTA, SCF, and PHD files, respectively. Also added `-trim_out',\n" );
  printf( "       which causes phred to write trimmed sequence and quality values\n" );
  printf( "       to all output files (FASTA, SCF, and PHD) except .poly files.\n" );
  printf( "       This is the same as using `-trim_fasta', `-trim_scf', and\n" );
  printf( "       `-trim_phd' in combination. See the section on `Sequence\n" );
  printf( "       Trimming' below.\n" );
  printf( "       NOTE: `-trim_phd' affects the values in the `TRIM' field of\n" );
  printf( "       the PHD file comment block.\n" );
  printf( "\n" );
  printf( "     - describe trimming in greater detail in the section `Sequence\n" );
  printf( "       Trimming'.\n" );
  printf( "\n" );
  printf( "     - add protection against buffer overflows in code\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "2. Acknowledgements.\n" );
  printf( "\n" );
  printf( "   Phred benefits from ideas developed by LaDeana Hillier, Mike Wendl,\n" );
  printf( "   Dave Ficenec, Tim Gleeson, Alan Blanchard, and Richard Mott.\n" );
  printf( "   \n" );
  printf( "\n" );
  printf( "3. Algorithms.\n" );
  printf( "\n" );
  printf( "   Phred uses simple Fourier methods to examine the four base traces in\n" );
  printf( "   the region surrounding each point in the data set in order to predict\n" );
  printf( "   a series of evenly spaced predicted locations.  That is, it determines\n" );
  printf( "   where the peaks would be centered if there were no compressions,\n" );
  printf( "   dropouts, or other factors shifting the peaks from their \"true\"\n" );
  printf( "   locations.\n" );
  printf( "\n" );
  printf( "   Next phred examines each trace to find the centers of the actual, or\n" );
  printf( "   observed, peaks and the areas of these peaks relative to their neighbors.\n" );
  printf( "   The peaks are detected independently along each of the four traces so\n" );
  printf( "   many peaks overlap.  A dynamic programming algorithm is used to match\n" );
  printf( "   the observed peaks detected in the second step with the predicted peak\n" );
  printf( "   locations found in the first step.\n" );
  printf( "\n" );
  printf( "   Phred evaluates the trace surrounding each called base using four or\n" );
  printf( "   five quality value parameters to quantify the trace quality.  It\n" );
  printf( "   uses a quality value lookup table to assign the corresponding quality\n" );
  printf( "   value.  The quality value is related to the base call error probability\n" );
  printf( "   by the formula\n" );
  printf( "\n" );
  printf( "     QV = - 10 * log_10( P_e )\n" );
  printf( "\n" );
  printf( "     where P_e is the probability that the base call is an error.\n" );
  printf( "\n" );
  printf( "   Phred uses data from a chemistry parameter file called 'phredpar.dat'\n" );
  printf( "   in order to identify dye primer data.  For dye primer data, phred\n" );
  printf( "   identifies loop/stem sequence motifs that tend to result in\n" );
  printf( "   CC and GG merged peak compressions.  It reduces the quality values\n" );
  printf( "   of potential merged peaks and splits those peaks that have certain\n" );
  printf( "   trace characteristics indicative of merged CC and GG peaks.  In\n" );
  printf( "   addition, the chemistry and dye information are passed to phrap.\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "4. Building and installing.\n" );
  printf( "\n" );
  printf( "   The INSTALL file describes the steps for building and installing\n" );
  printf( "   phred.\n" );
  printf( "\n" );
  printf( "   Copy the phred parameter file, called 'phredpar.dat', to a\n" );
  printf( "   directory that is accessible by phred users and set the environment\n" );
  printf( "   variable 'PHRED_PARAMETER_FILE' to the full path name of the file.\n" );
  printf( "   For example, if you copy 'phredpar.dat' to '/usr/local/etc/PhredPar'\n" );
  printf( "   and you are using the C shell then issue the command\n" );
  printf( "\n" );
  printf( "     %% setenv PHRED_PARAMETER_FILE /usr/local/etc/PhredPar/phredpar.dat\n" );
  printf( "\n" );
  printf( "   It is most convenient to set the environment variable in the system-\n" );
  printf( "   wide shell startup (cshrc or equivalent) file.\n" );
  printf( "\n" );
  printf( "   You can rename the phred parameter file but the PHRED_PARAMETER_FILE\n" );
  printf( "   environment variable must reflect the new name.\n" );
  printf( "\n" );
  printf( "   With Windows NT you give the command\n" );
  printf( "\n" );
  printf( "     %% set PHRED_PARAMETER_FILE=\\usr\\local\\etc\\PhredPar\\phredpar.dat\n" );
  printf( "\n" );
  printf( "   in the DOS command window in which you will run phred.\n" );
  printf( "\n" );
  printf( "   Note: if you compile phred on a SUN Solaris OS using the BSD C\n" );
  printf( "         compiler in the directory `/usr/ucb', you will find that\n" );
  printf( "         the `-id' command line option fails (phred reports that it\n" );
  printf( "         cannot read files, and it prints the name of each file it\n" );
  printf( "         fails to read; however, the name it prints lacks the first\n" );
  printf( "         few characters of the true name of the file). If this occurs,\n" );
  printf( "         recompile phred using either the optional C compiler in the\n" );
  printf( "         directory /opt/SUNWspro/bin or the GNU C compiler.\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "5. Running phred.\n" );
  printf( "\n" );
  printf( "   Phred uses command line options to control input, processing, and\n" );
  printf( "   output.  The command line options are delimited by a dash, \"-\".\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   The command line options are\n" );
  printf( "\n" );
  printf( "   Input Options\n" );
  printf( "   -------------\n" );
  printf( "   \n" );
  printf( "   -id <directory name>		Read and process files in <directory name>.\n" );
  printf( "\n" );
  printf( "   -if <file name>		Read and process files listed in the file\n" );
  printf( "                                <file name>.  Each line in <file name> must\n" );
  printf( "                                specify a valid path to a single input file.\n" );
  printf( "\n" );
  printf( "   -zd <directory name>         Location of compression program.  If -zd is\n" );
  printf( "                                omitted, phred uses the current path to search\n" );
  printf( "                                for the compression program.\n" );
  printf( "\n" );
  printf( "   -zt <directory name>         Directory where chromat is uncompressed. If\n" );
  printf( "                                -zd is omitted, phred uses /usr/tmp.  When\n" );
  printf( "                                phred processes a compressed file, it \n" );
  printf( "                                uncompresses the chromat into this temporary\n" );
  printf( "                                directory before it reads the file.  It\n" );
  printf( "                                subsequently deletes the uncompressed file in\n" );
  printf( "                                the temporary directory.\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   Processing Options\n" );
  printf( "   ------------------\n" );
  printf( "\n" );
  printf( "   -nocall			Disable phred base calling and set the\n" );
  printf( "                                current sequence to the ABI base calls\n" );
  printf( "                                that are read from the input file.  By\n" );
  printf( "                                default, the current sequence is set\n" );
  printf( "                                to the phred base calls.  This affects\n" );
  printf( "                                the base trimming and output options.\n" );
  printf( "\n" );
  printf( "   -trim <enzyme sequence>      Perform sequence trimming on the current\n" );
  printf( "                                sequence.  Bases are trimmed from the start\n" );
  printf( "                                and end of the sequence on the basis of\n" );
  printf( "                                trace quality.  In addition, <enzyme sequence>\n" );
  printf( "                                specifies a base sequence that is used\n" );
  printf( "                                to trim bases off the start of the current\n" );
  printf( "                                sequence.  You can specify a NULL enzyme\n" );
  printf( "                                sequence using empty double quotes, \"\".\n" );
  printf( "                                See the note below on the effect of using\n" );
  printf( "                                the trim option.\n" );
  printf( "\n" );
  printf( "   -trim_alt <enzyme sequence>  Perform sequence trimming on the current\n" );
  printf( "                                sequence.  Bases are trimmed from the start\n" );
  printf( "                                and end of the sequence on the basis of\n" );
  printf( "                                trace quality.  Specifically, for each base,\n" );
  printf( "                                the phred error probability is subtracted\n" );
  printf( "                                from the default value of 0.05 (or the value\n" );
  printf( "                                set using the `-trim_cutoff' option), and the\n" );
  printf( "                                resulting values are summed to find the\n" );
  printf( "                                maximum scoring subsequence.  Furthermore,\n" );
  printf( "                                the subsequence must have a minimum number\n" );
  printf( "                                of bases.  In addition, <enzyme sequence>\n" );
  printf( "                                specifies a base sequence that is used to\n" );
  printf( "                                trim bases off the start of the current\n" );
  printf( "                                sequence. You can specify a NULL enzyme\n" );
  printf( "                                sequence using empty double quotes, \"\".\n" );
  printf( "\n" );
  printf( "   -trim_cutoff <value>         Set trimming error probability for the\n" );
  printf( "                                `-trim_alt' option and the trimming points\n" );
  printf( "                                written in the phd files. The default value\n" );
  printf( "                                is 0.05.\n" );
  printf( "\n" );
  printf( "   -trim_fasta                  Trim sequences written to sequence and\n" );
  printf( "                                quality value FASTA files. Set trimming\n" );
  printf( "                                information in the FASTA headers to reflect\n" );
  printf( "                                the high quality of the sequence, and append\n" );
  printf( "                                the string `trimmed' to the header.\n" );
  printf( "\n" );
  printf( "   -trim_scf                    Trim sequence, quality values, and base\n" );
  printf( "                                locations written to SCF file. Append the\n" );
  printf( "                                string `trimmed' to the comments.\n" );
  printf( "\n" );
  printf( "   -trim_phd                    Trim sequence, quality values, and base\n" );
  printf( "                                locations written to PHD files. Also set the\n" );
  printf( "                                first and last high quality base locations\n" );
  printf( "                                specified in the `TRIM' comment field to\n" );
  printf( "                                the numbers of the first and last bases of\n" );
  printf( "                                the trimmed sequence (the first base in the\n" );
  printf( "                                sequence is base number zero). Finally set\n" );
  printf( "                                the error probability cutoff value in the\n" );
  printf( "                                `TRIM' comment field to -1.00 to indicate that\n" );
  printf( "                                the sequence is trimmed, and that the trim\n" );
  printf( "                                points may be unrelated to the error\n" );
  printf( "                                probability cutoff value.\n" );
  printf( "\n" );
  printf( "   -trim_out                    Trim information in the FASTA, SCF, and\n" );
  printf( "                                PHD output files. This is equivalent to\n" );
  printf( "                                specifying `-trim_fasta', `-trim_scf',\n" );
  printf( "                                and `-trim_phd' on the command line.\n" );
  printf( "\n" );
  printf( "   -nonorm                      Disable phred trace normalization.  This\n" );
  printf( "                                option is not recommended unless the base\n" );
  printf( "                                caller fails due to huge noise peaks\n" );
  printf( "                                extending over a large region at the start\n" );
  printf( "                                of the trace, as is characteristic of some\n" );
  printf( "                                dye terminator reactions.\n" );
  printf( "\n" );
  printf( "   -nosplit                     Disable compressed peak splitting.  By\n" );
  printf( "                                default, phred identifies and splits\n" );
  printf( "                                C and G peaks that may be a merged pair\n" );
  printf( "                                of peaks.  Phred searches for compression\n" );
  printf( "                                prone loop/stem sequence motifs and\n" );
  printf( "                                attempts to confirm a compression using\n" );
  printf( "                                characteristics of the trace, primarily\n" );
  printf( "                                the size of the candidate peak.\n" );
  printf( "\n" );
  printf( "   -nocmpqv                     Force phred to use the four parameter quality\n" );
  printf( "                                values.  By default, phred uses five parameter\n" );
  printf( "                                quality values for dye primer data (only) in\n" );
  printf( "                                order to reduce the quality values of merged\n" );
  printf( "                                CC and GG peaks.  (Phred uses the four\n" );
  printf( "                                parameter quality values for dye terminator\n" );
  printf( "                                chemistry data automatically.  If phred cannot\n" );
  printf( "                                determine the chemistry, it uses the four\n" );
  printf( "                                parameter quality values.)\n" );
  printf( "\n" );
  printf( "   -ceilqv <ceil_qv>            Specifies a maximum quality value assigned\n" );
  printf( "                                to bases.  Bases with quality value parameters\n" );
  printf( "                                that correspond to quality values greater\n" );
  printf( "                                than <ceil_qv> are assigned the value\n" );
  printf( "                                <ceil_qv>.\n" );
  printf( "\n" );
  printf( "   -beg_pred <trace_point>      Specifies the trace point at which to begin the\n" );
  printf( "                                peak prediction. This point should be in a\n" );
  printf( "                                region of `good' trace where the peak spacing\n" );
  printf( "                                is even and representative of the peak spacing\n" );
  printf( "                                throughout the trace. In addition the peaks\n" );
  printf( "                                should be large and the noise low in the\n" );
  printf( "                                region, and the value of <trace_point> must not\n" );
  printf( "                                be within 100 points of the trace ends.\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   Output Options\n" );
  printf( "   --------------\n" );
  printf( "\n" );
  printf( "   -st fasta                    Set the output sequence file format\n" );
  printf( "                                to FASTA. (Default.) Trimming options\n" );
  printf( "                                affect the FASTA file; see the Notes\n" );
  printf( "                                below for more information.\n" );
  printf( "\n" );
  printf( "   -st xbap                     Set the output sequence file format\n" );
  printf( "                                to XBAP.\n" );
  printf( "\n" );
  printf( "   -s                           Write sequence output files with the\n" );
  printf( "                                names obtained by appending \".seq\" to\n" );
  printf( "                                the names of the input files, and store\n" );
  printf( "                                them in the directory where phred is\n" );
  printf( "                                running.\n" );
  printf( "\n" );
  printf( "   -s <file name>               Write a sequence output file with the\n" );
  printf( "                                name <file name>.\n" );
  printf( "                                This option is valid for a single input\n" );
  printf( "                                file only.\n" );
  printf( "\n" );
  printf( "   -sd <directory name>         Write sequence output files with the\n" );
  printf( "                                names obtained by appending \".seq\" to\n" );
  printf( "                                the names of the input files, and write\n" );
  printf( "                                them in the directory <directory name>.\n" );
  printf( "\n" );
  printf( "   -sa <file name>              Write a sequence output file in FASTA\n" );
  printf( "                                format with the name <file name>.  The\n" );
  printf( "                                file contains the base calls of all the\n" );
  printf( "                                reads processed in this run of phred.\n" );
  printf( "\n" );
  printf( "   -qt fasta                    Set the output quality file format\n" );
  printf( "                                to FASTA. Trimming options affect the\n" );
  printf( "                                FASTA file; see the Notes below for\n" );
  printf( "                                more information.\n" );
  printf( "\n" );
  printf( "   -qt xbap                     Set the output quality file format\n" );
  printf( "                                to XBAP.  Trimmed off base quality\n" );
  printf( "                                values are omitted.\n" );
  printf( "\n" );
  printf( "   -qt mix                      Set the output quality file format\n" );
  printf( "                                to FASTA. Base quality values for\n" );
  printf( "                                all bases are written (including those\n" );
  printf( "                                for trimmed off bases).\n" );
  printf( "\n" );
  printf( "   -q                           Write quality output files with the\n" );
  printf( "                                names obtained by appending \".qual\" to\n" );
  printf( "                                the names of the input files, and store\n" );
  printf( "                                them in the directory where phred is\n" );
  printf( "                                running.\n" );
  printf( "                                This option is valid for FASTA format\n" );
  printf( "                                output files only.\n" );
  printf( "\n" );
  printf( "   -q <file name>               Write a quality output file with the\n" );
  printf( "                                name <file name>.\n" );
  printf( "                                This option is valid for a single input\n" );
  printf( "                                file and a FASTA format output file only.\n" );
  printf( "\n" );
  printf( "   -qd <directory name>         Write quality output files with the\n" );
  printf( "                                names obtained by appending \".qual\" to\n" );
  printf( "                                the names of the input files, and store\n" );
  printf( "                                them in the directory <directory name>.\n" );
  printf( "\n" );
  printf( "   -qa <file name>              Write a quality output file in FASTA\n" );
  printf( "                                format with the name <file name>.  The\n" );
  printf( "                                file contains the quality values of all the\n" );
  printf( "                                reads processed in this run of phred.\n" );
  printf( "\n" );
  printf( "   -qr <file name>              Write a histogram of the number of high\n" );
  printf( "                                quality bases per read.  This is meaning-\n" );
  printf( "                                ful when phred processes more than one\n" );
  printf( "                                read.\n" );
  printf( "\n" );
  printf( "   -c                           Write SCF files with the trace data,\n" );
  printf( "                                the base calls of the current sequences,\n" );
  printf( "                                and the positions of the base calls.  The\n" );
  printf( "                                SCF files have the names of the input\n" );
  printf( "                                files (phred will refuse to write the SCF\n" );
  printf( "                                file if you ask it to write the SCF file\n" );
  printf( "                                in the directory in which the input file\n" );
  printf( "                                resides).\n" );
  printf( "\n" );
  printf( "   -c <file name>               Write an SCF file with the trace data,\n" );
  printf( "                                the base calls of the current sequence,\n" );
  printf( "                                and the positions of the base calls.\n" );
  printf( "                                The SCF file has the name <file name>.\n" );
  printf( "                                This option is valid for a single input\n" );
  printf( "                                file only.\n" );
  printf( "\n" );
  printf( "   -cd <directory name>         Write SCF files with the trace data,\n" );
  printf( "                                the base calls of the current sequences,\n" );
  printf( "                                and the positions of the base calls.\n" );
  printf( "                                The SCF files are written in the directory\n" );
  printf( "                                <directory name> and have the same names\n" );
  printf( "                                as the input files.\n" );
  printf( "\n" );
  printf( "   -cp <number of bytes>        Store SCF trace data as 1 or 2 byte values.\n" );
  printf( "                                Defaults to 1 when the maximum trace value is\n" );
  printf( "                                less than 256, or to 2 when the maximum\n" );
  printf( "                                trace value is greater than or equal to 256.\n" );
  printf( "                                This is the trace precision.\n" );
  printf( "\n" );
  printf( "   -cs                          Always scale traces before writing them to\n" );
  printf( "                                an SCF output file. This ensures that the\n" );
  printf( "                                largest trace value has the largest value\n" );
  printf( "                                that can be stored in the SCF file. When the\n" );
  printf( "                                file trace precision is `1', the maximum\n" );
  printf( "                                value is 255, and when the precision is 2,\n" );
  printf( "                                the maximum value is 65535. Without this\n" );
  printf( "                                option, phred does not scale the trace unless\n" );
  printf( "                                (a) the trace was read from an ESD file or\n" );
  printf( "                                (b) the maximum trace value exceeds the value\n" );
  printf( "                                that can be stored in the SCF file at the\n" );
  printf( "                                precision used. Trace scaling ensures the\n" );
  printf( "                                maximum digital resolution for a given\n" );
  printf( "                                storage precision but it will make a\n" );
  printf( "                                uniformly low level trace appear to be a\n" );
  printf( "                                high level.\n" );
  printf( "\n" );
  printf( "   -p                           Write a PHD file, which is used by the\n" );
  printf( "                                consed editor to display bases.  A PHD\n" );
  printf( "                                file contains a set of comments used by\n" );
  printf( "                                consed for maintaining consistency between\n" );
  printf( "                                the chromat file, the .ace file and\n" );
  printf( "                                the PHD file, and it contains base data\n" );
  printf( "                                as triples consisting of the base call,\n" );
  printf( "                                quality, and position.  Phred always\n" );
  printf( "                                writes the first version of the PHD\n" );
  printf( "                                file for a read, which has the name\n" );
  printf( "                                <filename>.phd.1.  When a read is edited\n" );
  printf( "                                using consed, a new version of the phd is\n" );
  printf( "                                written by consed, for example, the second\n" );
  printf( "                                version has the name <filename>.phd.2.  With\n" );
  printf( "                                the -p option, <filename> is the name of the\n" );
  printf( "                                input file.\n" );
  printf( "\n" );
  printf( "   -p <filename>                Write a PHD file with the name <filename>.phd.1.\n" );
  printf( "                                This option is valid for processing a single\n" );
  printf( "                                input file.\n" );
  printf( "\n" );
  printf( "   -pd <directory name>         Write PHD files in directory <directory name>.\n" );
  printf( "                                The PHD files have the names <filename>.phd.1\n" );
  printf( "                                where <filename> is the name of the input file.\n" );
  printf( "\n" );
  printf( "   -d                           Write a data file that is used for detecting\n" );
  printf( "                                polymorphic bases.  The file has the\n" );
  printf( "                                name <filename>.poly where <filename> is the\n" );
  printf( "                                name of the input file.  The first line of\n" );
  printf( "                                the file consists of the sequence name, the\n" );
  printf( "                                smallest amplitude normalization factor, and\n" );
  printf( "                                the amplitude normalization factors for the\n" );
  printf( "                                A, C, G, and T traces.  One line for each\n" );
  printf( "                                called base follows the header line.  The\n" );
  printf( "                                information on each line consists of the\n" );
  printf( "                                called base, the position of the called base,\n" );
  printf( "                                the area of the called peak, the relative area\n" );
  printf( "                                of the called peak, the uncalled base, the\n" );
  printf( "                                position of the uncalled base, the area of the\n" );
  printf( "                                uncalled base, the relative area of the\n" );
  printf( "                                uncalled base, and the amplitudes of the four\n" );
  printf( "                                traces at the position of the called base.\n" );
  printf( "\n" );
  printf( "   -dd <dirname>                Write polymorphism data files in directory\n" );
  printf( "                                <directory name>.  The files have the names\n" );
  printf( "                                <filename>.poly where <filename> is the name\n" );
  printf( "                                of the input file.\n" );
  printf( "              \n" );
  printf( "   -raw <sequence name>         Write <sequence name> in the header of\n" );
  printf( "                                the sequence output file and the quality\n" );
  printf( "                                output file.\n" );
  printf( "                                By default, the name of the input file\n" );
  printf( "                                is written in the headers of these files.\n" );
  printf( "                                This option is valid for a single input\n" );
  printf( "                                file only.\n" );
  printf( "\n" );
  printf( "   -log                         Make phred append a log entry describing\n" );
  printf( "                                the processing run in the file \"phred.log\".\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   Miscellaneous\n" );
  printf( "   -------------\n" );
  printf( "\n" );
  printf( "   -v  <n>                      Verbose operation. You can control the level of\n" );
  printf( "                                verbosity with <n>, which ranges from 1 to 63.\n" );
  printf( "\n" );
  printf( "   -tags                        Label common output with tags in order to\n" );
  printf( "                                facilitate output parsing.\n" );
  printf( "\n" );
  printf( "   -h, -help                    Display a command line option summary.\n" );
  printf( "\n" );
  printf( "   -doc                         Display phred documentation.\n" );
  printf( "\n" );
  printf( "   -V                           Display phred version.                                \n" );
  printf( "\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   Examples\n" );
  printf( "   --------\n" );
  printf( "\n" );
  printf( "   If you plan to use phred base calls and base quality information as\n" );
  printf( "   input to the phrap assembly program and to the consed finishing\n" );
  printf( "   program, simply follow the documentation supplied with consed and\n" );
  printf( "   then type:\n" );
  printf( "\n" );
  printf( "   phredPhrap\n" );
  printf( "   \n" );
  printf( "   (with no arguments)\n" );
  printf( "\n" );
  printf( "   If you intend to use consed, you *MUST* use this perl script.  Failure\n" );
  printf( "   to use this script will result in many consed features not working\n" );
  printf( "   correctly, including consed's autofinish function, user-defined\n" );
  printf( "   consensus tags, tagging ALU and other repeats, and tagging vector\n" );
  printf( "   sequence.  Use the phredPhrap perl script.\n" );
  printf( "\n" );
  printf( "   An outline of the important processing steps performed by the script\n" );
  printf( "   follows.\n" );
  printf( "\n" );
  printf( "   Let us say you want to call bases from the chromat files in\n" );
  printf( "   subdirectory \"chromat_dir\", use phrap to assemble the contigs, and\n" );
  printf( "   run consed to edit/examine the contigs.  In this case you must ask\n" );
  printf( "   phred to create \"phd\" output files, which are required by consed.\n" );
  printf( "\n" );
  printf( "   It runs phred with the options\n" );
  printf( "\n" );
  printf( "     %% phred -id chromat_dir -pd phd_dir\n" );
  printf( "\n" );
  printf( "   which causes phred to read the chromat files in \"chromat_dir\" and\n" );
  printf( "   write the \"phd\" files to \"phd_dir\".  Next it makes FASTA files\n" );
  printf( "   from the \"phd\" files by running the phd2fasta program.\n" );
  printf( "   For example,\n" );
  printf( "\n" );
  printf( "     %% phd2fasta -id phd_dir -os seqs_fasta -oq seqs_fasta.screen.qual\n" );
  printf( "\n" );
  printf( "   Subsequently it screens out the vector in the sequences in\n" );
  printf( "   \"seqs_fasta\" using cross_match:\n" );
  printf( "\n" );
  printf( "     %% cross_match seqs_fasta vector.seq -minmatch 12 -minscore 20 -screen > screen.out\n" );
  printf( "\n" );
  printf( "   which generates the screened sequence file \"seqs_fasta.screen\",\n" );
  printf( "\n" );
  printf( "   It runs phrap to perform the sequence assembly as follows:\n" );
  printf( "\n" );
  printf( "     %% phrap seqs_fasta.screen -new_ace > phrap.out\n" );
  printf( "\n" );
  printf( "   Phrap writes the the assembled contigs to the file\n" );
  printf( "   \"seqs_fasta.screen.contigs\", and creates a .ace file that can be\n" );
  printf( "   used for importing the assembly to xbap, consed, or ace-mbly for\n" );
  printf( "   editing.\n" );
  printf( "\n" );
  printf( "   As another example, again you want to process the chromat files\n" );
  printf( "   in subdirectory \"chromat_dir\",  but now you want phred to write\n" );
  printf( "   the base calls to a FASTA file named \"seqs_fasta\" and the base\n" );
  printf( "   quality values to \"seqs_fasta.qual\".  In this case you run phred\n" );
  printf( "   with the options\n" );
  printf( "\n" );
  printf( "     %% phred -id chromat_dir -sa seqs_fasta -qa seqs_fasta.qual\n" );
  printf( "\n" );
  printf( "   We recommend that you not use the trim option.  Inaccurate bases\n" );
  printf( "   called near the ends of the traces will not interfere with proper\n" );
  printf( "   phrap assembly.\n" );
  printf( "\n" );
  printf( "   Refer to the file \"phrap.doc\", which is part of the phrap\n" );
  printf( "   distribution, for information on cross_match and phrap.\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   Return values\n" );
  printf( "   -------------\n" );
  printf( "\n" );
  printf( "   Phred returns 0 for successful processing and for file read errors. It\n" );
  printf( "   returns -1 for processing errors and file write errors.\n" );
  printf( "\n" );
  printf( "   Phred continues processing on file read and write errors but halts on\n" );
  printf( "   serious processing errors.\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "6. Phred parameter file\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   Phred reads the `primer ID' information in the chromatogram and it\n" );
  printf( "   tries to find the same name in the phred parameter file, which is\n" );
  printf( "   described in the `Building and installing' section above.  If it\n" );
  printf( "   succeeds, the phredpar.dat entry for the `primer ID' identifies the\n" );
  printf( "   sequencing reaction chemistry (primer or terminator) and the type of\n" );
  printf( "   dye.  If it cannot find the `primer ID' information in the\n" );
  printf( "   chromatogram, it reports\n" );
  printf( "\n" );
  printf( "     no dye primer ID in chromat yyyy\n" );
  printf( "\n" );
  printf( "   where yyyy is the chromatogram name. If it cannot find the `primer ID'\n" );
  printf( "   name in phredpar.dat (or it cannot find the phredpar.dat file), it\n" );
  printf( "   reports\n" );
  printf( "\n" );
  printf( "     unknown chemistry (xxxx) in chromat yyyy\n" );
  printf( "     add a line of the form\n" );
  printf( "     \"xxxx\"    <chemistry>      <dye type>      <machine type>\n" );
  printf( "     to the file zzzz\n" );
  printf( "     type `phred -doc' for more information\n" );
  printf( "\n" );
  printf( "   where xxxx is the `primer ID' and yyyy is the chromatogram name.\n" );
  printf( "   Add the indicated line to phredpar.dat.\n" );
  printf( "\n" );
  printf( "   Phred reads the `PHRED_PARAMETER_FILE' environment variable in order\n" );
  printf( "   to find the phredpar.dat file.  If this is not set on your system,\n" );
  printf( "   phred reports\n" );
  printf( "\n" );
  printf( "     warning: 'PHRED_PARAMETER_FILE' environment variable not set:\n" );
  printf( "               unable to identify chemistry and dye\n" );
  printf( "               type `phred -doc' for more information\n" );
  printf( "\n" );
  printf( "   If the `PHRED_PARAMETER_FILE' environment variable is set incorrectly,\n" );
  printf( "   that is, phred cannot find the phredpar.dat file there or the file is\n" );
  printf( "   not valid, phred reports\n" );
  printf( "\n" );
  printf( "     readParamFile: warning: unable to open file zzzz\n" );
  printf( "       warning: processing without phred parameters\n" );
  printf( "\n" );
  printf( "   where zzzz is the value of the PHRED_PARAMETER_FILE environment.\n" );
  printf( "   It processes the chromatograms but warns that it could not read\n" );
  printf( "   the parameter file as it processes each chromatogram as explained\n" );
  printf( "   above. In this case, you must set PHRED_PARAMETER_FILE to a valid name.\n" );
  printf( "\n" );
  printf( "   In these three cases phred processes the chromatogram but it uses the\n" );
  printf( "   default (ABI) four parameter quality values, does not try to split\n" );
  printf( "   compression peaks, and reports the chemistry and dye types to phrap\n" );
  printf( "   as `unknown'.\n" );
  printf( "\n" );
  printf( "   If you use a `primer ID' for your reactions that is not in phredpar.dat,\n" );
  printf( "   you can add the `primer ID' name to phredpar.dat.  You will need to know\n" );
  printf( "   the `primer ID' name as it is stored in the chromatograms, the chemistry\n" );
  printf( "   type (primer or terminator), the dye name, and the type of sequencing\n" );
  printf( "   machine.  Use a text editor to add `primer ID' entries to phredpar.dat.\n" );
  printf( "   You will find additional information about the form of phredpar.dat\n" );
  printf( "   entries in phredpar.dat.\n" );
  printf( "\n" );
  printf( "   The columns in phredpar.dat have the form\n" );
  printf( "\n" );
  printf( "   column      value name\n" );
  printf( "   ------      ----------\n" );
  printf( "   1           primer identification string\n" );
  printf( "   2           chemistry\n" );
  printf( "   3           dye\n" );
  printf( "   4           sequencing machine type\n" );
  printf( "\n" );
  printf( "   where the column values are separated by spaces or horizontal tabs.\n" );
  printf( "\n" );
  printf( "   The values phred recognizes are\n" );
  printf( "\n" );
  printf( "   value name                 values\n" );
  printf( "   ----------                 ------\n" );
  printf( "   primer id. string          primer name enclosed in double quotes\n" );
  printf( "   chemistry                  primer, terminator\n" );
  printf( "   dye                        rhodamine, d-rhodamine, big-dye,\n" );
  printf( "                              energy-transfer, bodipy\n" );
  printf( "   sequencing machine type    ABI_373_377, MolDyn_MegaBACE, ABI_3700\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   NOTE: the `MegaBACE Mobility File' entry in the phredpar.dat file\n" );
  printf( "         specifies `unknown' chemistry, rather than `primer' or\n" );
  printf( "         `terminator' because some early MegaBACE software wrote\n" );
  printf( "         `MegaBACE Mobility File' for the `primer ID' string in both\n" );
  printf( "         primer and terminator chemistry ABD files. You may want to\n" );
  printf( "         change this value if you process exclusively primer or\n" );
  printf( "         terminator chemistry MegaBACE data; however, you must\n" );
  printf( "         remember to change it if you decide to process different\n" );
  printf( "         chemistry data from the MegaBACE later.\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "7. Notes\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   Sequence Trimming\n" );
  printf( "   -----------------\n" );
  printf( "\n" );
  printf( "   First, a warning: do not trim sequences that phrap will assemble.  We\n" );
  printf( "   cautiously introduce trimming capabilities in phred to allow\n" );
  printf( "   identification of the high quality region of reads, and to permit\n" );
  printf( "   trimming off low quality segments of reads that are not destined for\n" );
  printf( "   a phrap assembly.\n" );
  printf( "\n" );
  printf( "   Phred uses a number of different algorithms to calculate trimming\n" );
  printf( "   information. The algorithm used and its effect depend on the output\n" );
  printf( "   file and the trimming-related command line options.\n" );
  printf( "\n" );
  printf( "   The phd output file always contains trimming information in the\n" );
  printf( "   header. Phred calculates this trimming information using a modified\n" );
  printf( "   Mott algorithm (it does not trim off vector sequence so the trimming\n" );
  printf( "   information identifies the entire high quality segment of the read).\n" );
  printf( "   The trimming information appears in the phd file header in the form\n" );
  printf( "\n" );
  printf( "   TRIM: <n1> <n2> <r1>\n" );
  printf( "\n" );
  printf( "   where <n1> is the first high quality base (where the first base in\n" );
  printf( "   the sequence is number zero) and <n2> is the last high quality base.\n" );
  printf( "   <r1> is the error probability cutoff value used to calculate the\n" );
  printf( "   trim points. The command line option `-trim_cutoff' affects the\n" );
  printf( "   phd file trimming information by setting the error probability cutoff\n" );
  printf( "   value used to calculate the base scores. If the sequence has fewer\n" );
  printf( "   than 20 high quality bases, the values <n1> and <n2> are set to -1.\n" );
  printf( "   If the `-trim_phd' or `-trim_out' option is used, <n1> and <n2>\n" );
  printf( "   are set to the numbers of the first and last bases in the trimmed\n" );
  printf( "   sequence (so <n1> is always zero), and <r1> is set to -1.00 to\n" );
  printf( "   indicate that the sequence is trimmed and that the error probability\n" );
  printf( "   cutoff value may be unrelated to the trim points.\n" );
  printf( "\n" );
  printf( "   The sequence, quality value, SCF, and PHD output files can be\n" );
  printf( "   affected by the trimming-related command line options. (Sequence\n" );
  printf( "   and quality value files are those created using the -s, -sa, -sd,\n" );
  printf( "   -q, -qa, and -qd options, SCF files are created using the -c and\n" );
  printf( "   -cd options, and PHD files are created using the -p and -pd\n" );
  printf( "   options). When phred runs without trimming-related options set,\n" );
  printf( "   it does not calculate trimming values for the sequence, quality\n" );
  printf( "   value, and SCF output files (and it does not `trim' the values\n" );
  printf( "   stored in them).\n" );
  printf( "\n" );
  printf( "   The `-trim' and `-trim_alt' options select the trimming algorithm\n" );
  printf( "   used to calculate the trimming information used in the sequence,\n" );
  printf( "   quality value, and SCF output files. The algorithm used for the\n" );
  printf( "   `-trim' option is based directly on characteristics of the trace.\n" );
  printf( "   It predates phred and phred quality values. The algorithm used for\n" );
  printf( "   the `-trim_alt' option is based on the modified Mott algorithm: it\n" );
  printf( "   uses the base error probabilities calculated from the phred quality\n" );
  printf( "   values and the error probability cutoff (the cutoff can be adjusted\n" );
  printf( "   using the -trim_cutoff option). We believe that the `-trim' option\n" );
  printf( "   tends to be conservative, `trimming off' more bases, in comparison to\n" );
  printf( "   the `-trim_alt' option. So we recommend using the `-trim_alt'\n" );
  printf( "   algorithm. Both the `-trim' and `-trim_alt' options take an argument\n" );
  printf( "   consisting of a vector sequence. If the argument is \"\" (null), phred\n" );
  printf( "   finds the high quality segment of the read. If the argument is not\n" );
  printf( "   null, and phred finds the beginning of the vector sequence within\n" );
  printf( "   the first 100 bases of the read, phred sets the left trim point to\n" );
  printf( "   remove the vector sequence as well as low quality bases.\n" );
  printf( "\n" );
  printf( "   Selecting either `-trim' or `-trim_alt' causes phred to determine\n" );
  printf( "   trimming information and to modify the sequence, quality value, and\n" );
  printf( "   SCF files as follows.\n" );
  printf( "\n" );
  printf( "     The FASTA sequence header contains trimming information\n" );
  printf( "     but the sequence is unaffected. The header has the form\n" );
  printf( "\n" );
  printf( "     >chromat_name   1323     15    548  ABI\n" );
  printf( "\n" );
  printf( "     where the sequence name immediately follows the header\n" );
  printf( "     delimiter, which is \">\", the first integer is the number\n" );
  printf( "     of bases called by phred, the second integer is the\n" );
  printf( "     number of bases `trimmed off' the beginning of the\n" );
  printf( "     sequence, the third integer is the number of bases\n" );
  printf( "     `remaining following trimming', and the string describes\n" );
  printf( "     the type of input file.\n" );
  printf( "\n" );
  printf( "     The XBAP-type of sequence header contains trimming\n" );
  printf( "     information, and the low quality bases are commented out.\n" );
  printf( "\n" );
  printf( "     For quality value file type option `-qt fasta' (default),\n" );
  printf( "     the FASTA quality value header contains the same trimming\n" );
  printf( "     information as in the FASTA sequence header and the\n" );
  printf( "     quality values of the `trimmed off' bases are set to zero.\n" );
  printf( "\n" );
  printf( "     For quality value file type option `-qt xbap', phred\n" );
  printf( "     writes a XBAP-type of sequence header with trimming\n" );
  printf( "     information followed by the quality values of the bases\n" );
  printf( "     remaining after trimming on subsequent lines.\n" );
  printf( "\n" );
  printf( "     For quality value file type option `-qt mix', phred\n" );
  printf( "     writes a FASTA quality value header with the same\n" );
  printf( "     trimming information as in the FASTA sequence header\n" );
  printf( "     followed by the quality values of all bases (without\n" );
  printf( "     trimming).\n" );
  printf( "\n" );
  printf( "     The SCF file contains trimming information in the header,\n" );
  printf( "     and the sequence, quality values, and trace locations of\n" );
  printf( "     the called peaks are unaffected. The left clip is the\n" );
  printf( "     number of bases to trim off the left end of the sequence\n" );
  printf( "     and the right clip is the number of bases to trim off\n" );
  printf( "     the right end.\n" );
  printf( "     \n" );
  printf( "   When the `-trim_fasta' or `-trim_out' option is used with the `-trim'\n" );
  printf( "   or `-trim_alt' (and -s, -sa, -sd, -q, -qa, or -qd) option, phred\n" );
  printf( "   writes the trimmed sequence to the sequence FASTA file and trimmed\n" );
  printf( "   quality values to the quality value FASTA file; that is, it writes\n" );
  printf( "   only the high quality bases and the corresponding quality values. In\n" );
  printf( "   addition, it appends the string `trimmed' to the FASTA headers and\n" );
  printf( "   the trimming information in the header indicates that no (additional)\n" );
  printf( "   bases are to be trimmed off. The option `-trim_fasta' is invalid with\n" );
  printf( "   the `-qt xbap' and `-qt mix' options.\n" );
  printf( "\n" );
  printf( "   When the `-trim_scf' or `-trim_out' option is used with the `-trim'\n" );
  printf( "   or `-trim_alt' (and -c or -cd) option, phred writes the trimmed\n" );
  printf( "   sequence, trimmed quality value, and trimmed called peak locations to\n" );
  printf( "   the SCF output file. In addition, it appends the string `trimmed' to\n" );
  printf( "   the comment field and the left and right clip values are set to zero.\n" );
  printf( "\n" );
  printf( "   When the `-trim_phd' or `-trim_out' option is used with the `-trim'\n" );
  printf( "   or `-trim_alt' (and -p or -pd) option, phred writes the trimmed\n" );
  printf( "   sequence, trimmed quality value, and trimmed called peak locations to\n" );
  printf( "   the PHD output file. In addition, when it writes the `TRIM' field\n" );
  printf( "   in the comment block (at the beginning of the file), it sets the\n" );
  printf( "   values for the first and last high quality bases to the numbers of\n" );
  printf( "   the first and last bases of the trimmed sequence (where the first\n" );
  printf( "   base is number zero), and it sets the error probability cutoff value\n" );
  printf( "   to -1.00. Setting the cutoff value to -1.00 indicates that the\n" );
  printf( "   sequence is trimmed, and that the trim points may be unrelated to the\n" );
  printf( "   error probability cutoff value.\n" );
  printf( "\n" );
  printf( "   The modified Mott trimming algorithm, which is used to calculate the\n" );
  printf( "   trimming information for the `-trim_alt' option and the phd files,\n" );
  printf( "   uses base error probabilities calculated from the phred quality\n" );
  printf( "   values. For each base it subtracts the base error probability from an\n" );
  printf( "   error probability cutoff value (0.05 by default, and changed using\n" );
  printf( "   the `-trim_cutoff' option) to form the base score. Then it finds the\n" );
  printf( "   highest scoring segment of the sequence where the segment score is\n" );
  printf( "   the sum of the segment base scores (the score can have non-negative\n" );
  printf( "   values only). The algorithm requires a minimum segment length, which\n" );
  printf( "   is set to 20 bases.\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   ESD Files\n" );
  printf( "   ---------\n" );
  printf( "\n" );
  printf( "   Phred reads processed MegaBACE ESD files.  It cannot read the raw\n" );
  printf( "   ESD files.  It is important that you identify the dye chemistry\n" );
  printf( "   correctly when you run the MegaBACE base caller so that phred can\n" );
  printf( "   assign the right base to each trace. (This is important with ABI\n" );
  printf( "   data too.)\n" );
  printf( "\n" );
  printf( "   In order to obtain the best phred quality value accuracy with\n" );
  printf( "   MegaBACE data, phred must use the quality value lookup tables\n" );
  printf( "   designed for this data.  Phred identifies the sequencing machine\n" );
  printf( "   by reading the `primer ID' string in the chromatogram and matching\n" );
  printf( "   it with an entry in the phredpar.dat file.  The matching entry\n" );
  printf( "   lists the chemistry, dye, and sequencing machine types. For example,\n" );
  printf( "   the `primer ID' string of the form `ET Primer' identifies a\n" );
  printf( "   chromatogram as ET dye primer data generated on a MegaBACE\n" );
  printf( "   sequencing machine. You can check that phred interprets the\n" );
  printf( "   `primer ID' string correctly by using the `-v 63' option to have\n" );
  printf( "   phred write diagnostic information to the screen.\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "   LI-COR Data\n" );
  printf( "   -----------\n" );
  printf( "\n" );
  printf( "   Band Spread Ratio (BSR)\n" );
  printf( "\n" );
  printf( "   Phred reads SCF files created by the LI-COR gel processing software\n" );
  printf( "   and has quality value lookup tables calibrated for traces processed\n" );
  printf( "   with Band Spread Ratio (BSR) of 2.2. The LI-COR software writes a\n" );
  printf( "   `primer ID' string in the SCF file that indicates the BSR value\n" );
  printf( "   used in the trace processing, which for BSR=2.2 is \n" );
  printf( "   `DyePrimer{LI-COR_IR_2.2}'. Accordingly, the phredpar.dat file in\n" );
  printf( "   this distribution has an entry with this string, which enables\n" );
  printf( "   phred to recognize LI-COR traces processed with BSR=2.2, and to\n" );
  printf( "   use the quality value lookup table designed for this LI-COR data.\n" );
  printf( "   Phred has a quality value lookup table for data processed with\n" );
  printf( "   BSR=2.2 only so the quality values for LI-COR traces processed\n" );
  printf( "   with other BSR values will have reduced accuracy.\n" );
  printf( "\n" );
  printf( "\n" );
  printf( "8. References\n" );
  printf( "\n" );
  printf( "   Brent Ewing, LaDeana Hillier, Michael C. Wendl, and Phil Green.\n" );
  printf( "   Base-calling of automated sequencer traces using phred. I. Accuracy\n" );
  printf( "   assessment. 1998. Genome Research 8:175-185.\n" );
  printf( "\n" );
  printf( "   Brent Ewing and Phil Green\n" );
  printf( "   Base-calling of automated sequencer traces using phred. II. Error\n" );
  printf( "   probabilities. 1998. Genome Research 8:186-194.\n" );
  printf( "\n" );
  printf( "\n" );

  return( 0 );
}
